### PR TITLE
Implement liveness probes for indexer and janitor service

### PR DESCRIPTION
**Pull request by guilload**
### Description
Implement basic liveness probes for indexer and janitor service

### How was this PR tested?
Ran quickwit and killed the metastore while  `curl`ing the livness endpoint.

### CODE REVIEW

**Comment by guilload** [comment]
The probe endpoints being behind the `api/v1` prefix is a bit awkward. Shall I fix that?

**Comment by fmassot** [comment]
> The probe endpoints being behind the `api/v1` prefix is a bit awkward. Shall I fix that?

Yes, good point, it was stupid to put them here.

File: quickwit/quickwit-indexing/src/actors/indexing_service.rs [hunk 1]
```diff
@@ -156,7 +151,7 @@ impl IndexingService {
             storage_resolver,
             local_split_store: Arc::new(local_split_store),
             indexing_pipeline_handles: Default::default(),
-            state: Default::default(),
+            counters: Default::default(),
```
Comment (fmassot) [inline quickwit/quickwit-indexing/src/actors/indexing_service.rs:154]: You can't resist, it's too strong :)

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 2]
```diff
@@ -86,7 +86,7 @@ async fn run_postgres_migrations(pool: &Pool<Postgres>) -> MetastoreResult<()> {
         tx.rollback().await?;
         error!(err=?migration_err, "Database migrations failed");
         return Err(MetastoreError::InternalError {
-            message: "Failed to run migrator on Postgresql database.".to_string(),
+            message: "Failed to run migration on Postgresql database.".to_string(),
```
Comment (fmassot) [inline quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:89]: nice catch :)

File: quickwit/quickwit-serve/src/rest.rs [hunk 3]
```diff
@@ -37,15 +39,19 @@ use crate::search_api::{search_get_handler, search_post_handler, search_stream_h
...
     rest_listen_addr: SocketAddr,
+    universe: &Universe,
     quickwit_services: &QuickwitServices,
 ) -> anyhow::Result<()> {
     info!(rest_listen_addr = %rest_listen_addr, "Starting REST server.");
     let request_counter = warp::log::custom(|_| {
         crate::SERVE_METRICS.http_requests_total.inc();
     });
+
+    let delete_task_service_opt: Option<Mailbox<DeleteTaskService>> = universe.get_one();
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/rest.rs:53]: it's a bit weird to have this here alone, but I see the rationale: `delete_task_service_opt` is not a service as important as the others. So you did not include it in the health checks right? Thus this `get_one`.

Comment (fmassot) [reply quickwit/quickwit-serve/src/rest.rs:53]: and now we also return a janitor service so it's harder to get this delete service. okokok

**Pull request merged**